CMakeLists.txt

```diff
@@ -1,5 +1,32 @@
 cmake_minimum_required()
 
+project(QuantLib)
+
+include(${CMAKE_CURRENT_LIST_DIR}/cmake/quantlib.cmake)
+
+if (MSVC)
+    set(MSVC_RUNTIME "static" CACHE STRING "MSVC runtime to link")
+    set_property(CACHE MSVC_RUNTIME PROPERTY STRINGS static dynamic)
+    option(USE_BOOST_DYNAMIC_LIBRARIES "Use the shared version of Boost libraries" ON)
```

Comment:
This option, and the corresponding definition of `-DBOOST_TEST_DYN_LINK`, are also needed on Linux.  The option can be moved out of the `if (MSVC)` and the definition can be added back to `test-suite/CMakeLists.txt`.

Reply:
I'll move it outside, yes.

About moving it back to `test-suite/CMakeLists.txt`, don't you think it is better at root level? If Quantlib starts to depend on any boost binary, then it will work for all of them (not just for unittest lib).

Reply:
I agree, `-DBOOST_DYN_LINK` goes at root level.
On the other hand, `-DBOOST_TEST_DYN_LINK` is specific and can go in the test-suite folder.